CMakeLists.txt

```diff
@@ -24,13 +13,10 @@ string(COMPARE EQUAL ${CMAKE_CURRENT_SOURCE_DIR} ${CMAKE_SOURCE_DIR} PROJECT_IS_
 option(GSL_INSTALL "Generate and install GSL target" ${PROJECT_IS_TOP_LEVEL})
 option(GSL_TEST "Build and perform GSL tests" ${PROJECT_IS_TOP_LEVEL})
 
-# This GSL implementation generally assumes a platform that implements C++14 support.
-set(gsl_min_cxx_standard "14")
-
-if (PROJECT_IS_TOP_LEVEL)
-    gsl_set_default_cxx_standard(${gsl_min_cxx_standard})
-else()
-    gsl_client_set_cxx_standard(${gsl_min_cxx_standard})
+# Because GSL is a header only library, that assumes at minimum C++14 support
+# We must require clients to use a compiler that supports at least that.
+if (CMAKE_VERSION VERSION_GREATER "3.8" OR CMAKE_VERSION VERSION_EQUAL "3.8")
+    target_compile_features(GSL INTERFACE "cxx_std_14")
 endif()
```

Comment:
To summarize the solution.

If CMake less than 3.8 do nothing. This leaves it up to the user which is call we can do.
If CMake greater than or equal to 3.8, use compile feature cxx_std_14. To ensure that users of the library are compiling GSL as expected/required.